project/BUILD

```diff
@@ -23,13 +23,33 @@ java_library(
     deps = ["//third_party/java/protobuf:protobuf"],
 )
 
-java_plugin(
-    name = "dagger_query_plugin",
-    srcs = ["src/com/google/daggerquery/plugin/QueryPlugin.java"],
+java_library(
+    name = "dagger_query_plugin_sources",
+    srcs = glob(["src/com/google/daggerquery/plugin/**/*.java"]),
```

Comment:
The convention in Google is to have a BUILD file in each folder that has sources (https://docs.bazel.build/versions/master/build-ref.html#packages), so that you never have to make cross-folder references in srcs. This results in better modularity.

So in your case it would be something like:
src/com/google/daggerquery/BUILD
src/com/google/daggerquery/plugin/BUILD
tests/com/google/daggerquery/BUILD
tests/com/google/daggerquery/plugin/BUILD
etc.